streams/src/test/java/org/apache/kafka/streams/kstream/internals/CogroupedKStreamImplTest.java

```diff
@@ -173,6 +174,56 @@ public void shouldNameProcessorsAndStoreBasedOnNamedParameter() {
                 "      <-- KTABLE-TOSTREAM-0000000005\n\n"));
     }
 
+    @Test
+    public void shouldNameProcessorsAndStoreBasedOnNamedParameterRepartition() {
```

Comment:
I don't understand the test name? Don't we test if a repartition topic is inserted? `shouldInsertRepartitionsTopicForUpstreamKeyModification()`

Can we add an actual integration test using `EmbeddedKafkaCluster` -- we don't really need to pipe data, but should just verify that the topics are created with the correct number of partitions. TTD cannot help to verify the number of partitions.

Reply:
Should we also extend `RepartitionOptimizingTest` ?